C23_WMDE_Mobile_DE_10

Sperling-0 opened the pull request:
Based on VAR of test 09
VAR uses informal form of text

Ticket: https://phabricator.wikimedia.org/T351696

Sperling-0 committed:
Archive C23_WMDE_Mobile_DE_09
Ticket: https://phabricator.wikimedia.org/T351696

Sperling-0 committed:
Prepare campaign for C23_WMDE_Mobile_DE_10
Based on VAR of test 09

Ticket: https://phabricator.wikimedia.org/T351696

Abban commented on test/banners/mobile_english/components/BannerVar.spec.ts:
```diff
@@ -22,7 +22,7 @@ let pageScroller: PageScroller;
 const formModel = useFormModel();
 const translator = ( key: string ): string => key;
 
-describe( 'BannerVar.vue', () => {
```
Looks like there was a few changes caused by a PhpStorm refactor. You'll need to revert these.

Sperling-0 replied:
Done.

Sperling-0 committed:
Implement VAR for C23_WMDE_Mobile_DE_10
VAR uses informal form of text

Ticket: https://phabricator.wikimedia.org/T351696

gbirke commented on banners/mobile/messages.ts:
```diff
@@ -22,7 +22,7 @@ const messages: TranslationMessages = {
 		'per E-Mail. Sie erhalten von uns keine Information, wenn Wikipedia wieder Hilfe braucht.',
 	'soft-close-prompt': 'Wikipedia später unterstützen?',
-	'use-of-funds-link': 'Was Ihre Spende bewirkt',
+	'use-of-funds-link': 'Deine Spende wirkt',
```
This will change the link in both banners to informal (while the CTRL stay formal entirely). The easiest way to recitify this will probably be a `messages_var.ts` included in the entry point

Sperling-0 replied:
Done.

gbirke reviewed (changes requested):
The mini banner still shows a formal message, "Ist Ihnen Wikipedia 10&nbsp;€ wert" instead of "Ist dir Wikipedia 10&nbsp;€ wert"  The easiest way to change this would be to create a var version of MiniBanner as well.

A better/different way to do this would be replace the element `wmde-banner-mini-headline` in the MiniBanner with a named slot, move the headline element into its own content component (duplicated for this test) and pass different contents into that slot in `BannerCtrl` and `BannerVar`. What do you think, @Abban - is that extensibility worth it (because there might be tests in the future where the the campaigns team wants to test different messages or remove the header again) or is it too speculative and duplication (and cleanup in the next test) is the way to go?

Abban commented on banners/mobile/content/BannerSlidesVar.vue:
```diff
@@ -0,0 +1,66 @@
+
+	<KeenSliderSlide :is-current="currentSlide === 3">
+		<p>
+			{{ visitorsVsDonorsSentence }} Die meisten Menschen spenden, weil sie Wikipedia nützlich finden.
```
You should add the highlighting to all the `visitorsVsDonorsSentence` instances. Have a look in the archived banner you can see it wrapped in an `AnimatedText` component.

Sperling-0 replied:
Done.

Abban commented on banners/mobile/content/BannerTextVar.vue:
```diff
@@ -0,0 +1,36 @@
+	visitorsVsDonorsSentence
+}: DynamicContent = inject( 'dynamicCampaignText' );
+
+const currentDate = computed( () => {
```
I think you don't need this as the dunamicCampaignText will provide it. Eg:

```js
const {
	currentDayName,
    currentDate,
	visitorsVsDonorsSentence
}: DynamicContent = inject( 'dynamicCampaignText' );
```

Sperling-0 replied:
Done.

Abban replied:
Can you also do the same in the Slides components?

Sperling-0 replied:
Done.

Abban commented on test/banners/mobile/components/BannerCtrl.spec.ts:
```diff
@@ -75,15 +74,6 @@ describe( 'BannerCtrl.vue', () => {
 		vi.useRealTimers();
 	} );
 
-	describe( 'Content', () => {
```
These tests make sure the animated `visitorsVsDonors` sentence is in the banners, can you put them back?

Please also add `expectHidesAnimatedVisitorsVsDonorsSentenceInMessage` and `expectHidesAnimatedVisitorsVsDonorsSentenceInSlideShow` to the data provider array.

Sperling-0 replied:
Done.

Abban commented on test/banners/mobile/components/BannerVar.spec.ts:
```diff
@@ -1,6 +1,6 @@
 import { afterEach, beforeEach, describe, test, vi } from 'vitest';
 import { mount, VueWrapper } from '@vue/test-utils';
-import Banner from '../../../../banners/mobile/components/BannerVar.vue';
+import Banner from '../../../../banners/mobile/components/BannerCtrl.vue';
```
This should import `BannerVar.vue`

Sperling-0 replied:
Done.

Abban reviewed (changes requested):
Mostly there, you just need to make sure the animated text is there and is tested.

Abban commented:
@gbirke 

>is that extensibility worth it 

I would use a MiniBannerVar in this instance, and maybe next time they want to test make it a slot at that point.
